File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int max=0,min=INT_MAX,curPro=0;
        for(int sellIdx=0;sellIdx<n;sellIdx++){
            min = fmin(prices[sellIdx],min);
       
            curPro = prices[sellIdx] - min;
            cout<<min<<" "<<curPro;

            max = fmax(max,curPro);
            cout<<max;
        }
        return max;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = Number.MAX_VALUE;
    let max = 0;
    
    for(let i = 0;i<prices.length;i++){
        if(min > prices[i]){
            min = prices[i];
        }else{
            max = Math.max(max, prices[i]-min);
        }
    }
    return max;
};
